File: servNatMan/routers/games.py
from fastapi import APIRouter, HTTPException, Depends
import psycopg2
from database import get_db_connection
from models import GameCreate, GameResponse

router = APIRouter(prefix="/games", tags=["games"])

@router.get("/", response_model=dict)
async def get_games(conn = Depends(get_db_connection)):
    with conn.cursor() as cur:
        cur.execute("SELECT * FROM games")
        games = cur.fetchall()
    return {"games": games}

@router.post("/", response_model=dict)
async def create_game(game: GameCreate, conn = Depends(get_db_connection)):
    with conn.cursor() as cur:
        try:
            cur.execute("""
                INSERT INTO games (start_time, end_time, id_region)
                VALUES (%s, %s, %s) RETURNING id
            """, (game.start_time, game.end_time, game.id_region))
            
            new_game_id = cur.fetchone()['id']
            conn.commit()
            return {"message": "Игра создана", "game_id": new_game_id}
            
        except Exception as e:
            conn.rollback()
            raise HTTPException(status_code=500, detail=f"Ошибка: {str(e)}")

File: NatManApp/src/services/games.js
import axios from 'axios';

const API_URL = 'http://217.114.14.77:8002/games';

export async function getGames() {
    try {
        const response = await axios.get(API_URL);
        return response.data.games; // массив игр из БД
    } catch (error) {
        console.error("Ошибка при получении игр:", error);
        return null;
    }
}

export async function createGame(startTime, endTime, idRegion) {
    try {
        const response = await axios.post(API_URL, {
            start_time: startTime,
            end_time: endTime,
            id_region: idRegion
        });
        return response.data;  // успех / созданная игра
    } catch (error) {
        console.error("Ошибка при создании игры:", error);
        return null;
    }
}